business_requirement_deliverable/models/business.py

```diff
@@ -0,0 +1,160 @@
+    def _get_price_total(self):
+        self.price_total = self.unit_price * self.qty
+
+    @api.one
+    @api.onchange('product_id')
+    def product_id_change(self):
+        description = ''
+        uom_id = False
+        unit_price = 0
+        product = self.product_id
+        if product:
+            description = product.name
+            uom_id = product.uom_id.id
+            unit_price = product.standard_price
+        self.description = description
+        self.uom_id = uom_id
+        self.unit_price = unit_price
+
+    @api.one
+    @api.onchange('resource_type')
+    def resource_type_change(self):
+        if self.resource_type == 'procurement':
+            self.user_id = False
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _name = "business.requirement.deliverable"
+    _description = "Business Requirement Deliverable"
+
+    sequence = fields.Integer('Sequence')
+    description = fields.Text('Description', required=True)
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=False
+    )
+    uom_id = fields.Many2one(
+        comodel_name='product.uom',
+        string='UoM',
+        required=True
+    )
+    qty = fields.Integer(
+        string='Quantity',
+        store=True,
+        default=1,
+    )
+    resource_ids = fields.One2many(
+        comodel_name='business.requirement.resource',
+        inverse_name='business_requirement_deliverable_id',
+        string='Business Requirement Resource',
+        copy=False,
+    )
+    business_requirement_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Business Requirement',
+        ondelete='cascade'
+    )
+    unit_price = fields.Float(
+        string='Unit Price'
+    )
+    price_total = fields.Float(
+        compute='_get_price_total',
+        string='Subtotal'
+    )
+
+    @api.one
+    @api.depends('unit_price', 'qty')
+    def _get_price_total(self):
+        self.price_total = self.unit_price * self.qty
+
+    @api.one
+    @api.onchange('product_id')
+    def product_id_change(self):
+        description = ''
+        uom_id = False
+        unit_price = 0
+        product = self.product_id
+        if product:
+            description = product.name
+            uom_id = product.uom_id.id
+            unit_price = product.standard_price
+        self.description = description
+        self.uom_id = uom_id
+        self.unit_price = unit_price
+
+
+class BusinessRequirement(models.Model):
+    _inherit = "business.requirement"
+
+    deliverable_lines = fields.One2many(
+        comodel_name='business.requirement.deliverable',
+        inverse_name='business_requirement_id',
+        string='Deliverable Lines',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    resource_cost_total = fields.Float(
+        compute='_get_deliverable_cost_total',
+        string='Total Price',
```

Comment:
agree
Total Cost is the sum of all resources linked to the BR
Add the following:
Total Price is the sum of all deliverable linked to the BR
Not sure we should have function "Gross Profit"
Actually @dreispt @jbficent I am not sure whether it is a good idea to have it here because some consultants might work side by side with the customer on the BR and this might show some sensitive information.
Ideally (maybe on the roadmap), I'd rather have a specific menu to handle the resources entry separately.
* One menu allows the BR only
* One menu allows the BR+deliverables only
* One menu allows the BR+deliverables+resources
It doesnot impact models, only menus and forms